core/base/src/TSystem.cxx

```diff
@@ -3340,7 +3372,12 @@ int TSystem::CompileMacro(const char *filename, Option_t *opt,
          };
          ForeachSharedLibDep(lib, LoadLibF);
       }
-      return !gSystem->Load(lib);
+      int res;
+      {
+         R__LOCKGUARD(gGlobalMutex); // NEWLY_ADDED
```

Comment:
`TSystem::Load` should be thread-safe on  its own.
(Albeit if another thread is detroying/recreating the library, this might be a problem :( ).